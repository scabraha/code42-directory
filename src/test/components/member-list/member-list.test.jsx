import React from 'react';
import renderer from 'react-test-renderer';
import MemberList from '../../../components/member-list/member-list';

describe('Member List', () => {
  it('Renders the view with no team members and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberList
        teamMembers={[]}
        onMemberSelect={() => {}}
        selectedMemberId=''
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with team members and no selection and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberList
        teamMembers={[
          { login: 'test1', avatar_url: 'url1' },
          { login: 'test2', avatar_url: 'url2' },
          { login: 'test3', avatar_url: 'url3' }
        ]}
        onMemberSelect={() => {}}
        selectedMemberId=''
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with team members and a selection and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberList
        teamMembers={[
          { login: 'test1', avatar_url: 'url1' },
          { login: 'test2', avatar_url: 'url2' },
          { login: 'test3', avatar_url: 'url3' }
        ]}
        onMemberSelect={() => {}}
        selectedMemberId='test2'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

