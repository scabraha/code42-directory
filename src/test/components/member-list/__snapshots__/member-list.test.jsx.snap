// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member List Renders the view with no team members and ensures it renders correctly 1`] = `
<div
  className="member-view"
>
  <ul />
</div>
`;

exports[`Member List Renders the view with team members and a selection and ensures it renders correctly 1`] = `
<div
  className="member-view"
>
  <ul>
    <li
      className="member-list-item noselect"
      onClick={[Function]}
    >
      <img
        alt=""
        src="url1"
      />
      <div>
        test1
      </div>
    </li>
    <li
      aria-selected={true}
      className="member-list-item noselect"
      onClick={[Function]}
    >
      <img
        alt=""
        src="url2"
      />
      <div>
        test2
      </div>
    </li>
    <li
      className="member-list-item noselect"
      onClick={[Function]}
    >
      <img
        alt=""
        src="url3"
      />
      <div>
        test3
      </div>
    </li>
  </ul>
</div>
`;

exports[`Member List Renders the view with team members and no selection and ensures it renders correctly 1`] = `
<div
  className="member-view"
>
  <ul>
    <li
      className="member-list-item noselect"
      onClick={[Function]}
    >
      <img
        alt=""
        src="url1"
      />
      <div>
        test1
      </div>
    </li>
    <li
      className="member-list-item noselect"
      onClick={[Function]}
    >
      <img
        alt=""
        src="url2"
      />
      <div>
        test2
      </div>
    </li>
    <li
      className="member-list-item noselect"
      onClick={[Function]}
    >
      <img
        alt=""
        src="url3"
      />
      <div>
        test3
      </div>
    </li>
  </ul>
</div>
`;
